### PR TITLE
Configure Spring Security

### PR DESCRIPTION
**Pull request by KostiantynVorobiov**

### CODE REVIEW

File: src/main/java/com/cinema/congig/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Autowired
+    public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception {
+        auth
+                .inMemoryAuthentication()
+                .passwordEncoder(getEncoder())
+                .withUser("bob").password(getEncoder().encode("12345")).roles("USER");
```
Comment (AnaLanda) [inline src/main/java/com/cinema/congig/SecurityConfig.java]:
```suggestion
                .withUser("bob")
                .password(getEncoder().encode("12345"))
                .roles("USER");
```

File: src/main/java/com/cinema/controller/CustomGlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        Map<String, Object> body = new HashMap<>();
+        body.put("timestamp", new Date());
+        body.put("status", status.value());
+        List<String> errors = ex.getBindingResult()
+                .getAllErrors()
+                .stream()
+                .map(DefaultMessageSourceResolvable::getDefaultMessage)
+                .collect(Collectors.toList());
+        body.put("errors", errors);
+
```
Comment (AnaLanda) [inline src/main/java/com/cinema/controller/CustomGlobalExceptionHandler.java]: You can remove the empty line here

**Review by AnaLanda — COMMENTED** [review]
Looks good 😃

File: src/main/java/com/cinema/controller/CustomGlobalExceptionHandler.java (hunk 2 above)
Comment (RomanDubovskyi) [reply src/main/java/com/cinema/controller/CustomGlobalExceptionHandler.java]: > You can remove the empty line here

\+

File: src/main/java/com/cinema/controller/OrderController.java [hunk 3]
```diff
@@ -33,15 +34,20 @@ public OrderController(OrderService orderService, UserService userService,
     }
 
     @GetMapping
-    public List<OrderResponseDto> getOrderHistory(@RequestParam Long userId) {
-        return orderService.getOrderHistory(userService.get(userId)).stream()
+    public List<OrderResponseDto> getOrderHistory(Authentication authentication) {
+        UserDetails principal = (UserDetails) authentication.getPrincipal();
+        String email = principal.getUsername();
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/controller/OrderController.java]: can we user authentication.getName() instead?

File: src/main/java/com/cinema/model/dto/CinemaHallRequestDto.java [hunk 4]
```diff
@@ -1,9 +1,15 @@
...
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Positive;
 import lombok.Data;
 
 @Data
 public class CinemaHallRequestDto {
+    @Positive(message = "Must be greater than 0")
     private int capacity;
+    @NotNull(message = "Should be description for cinema hall")
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/model/dto/CinemaHallRequestDto.java]:
```suggestion
    @NotNull(message = "Description can't be null")
```
write informative messages

File: src/main/java/com/cinema/model/dto/MovieRequestDto.java [hunk 5]
```diff
@@ -1,9 +1,12 @@
 package com.cinema.model.dto;
 
+import javax.validation.constraints.NotEmpty;
 import lombok.Data;
 
 @Data
 public class MovieRequestDto {
+    @NotEmpty(message = "Should be movie title")
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/model/dto/MovieRequestDto.java]:
```suggestion
    @NotEmpty(message = "Movie title can't be empty")
```

File: src/main/java/com/cinema/model/dto/MovieSessionRequestDto.java [hunk 6]
```diff
@@ -1,10 +1,15 @@
 package com.cinema.model.dto;
 
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.Positive;
 import lombok.Data;
 
 @Data
 public class MovieSessionRequestDto {
+    @Positive(message = "Can be numbers greater then 0")
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/model/dto/MovieSessionRequestDto.java]: same here

File: src/main/java/com/cinema/validator/ValidPassword.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+@Documented
+@Constraint(validatedBy = PasswordValidator.class)
+@Target({ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface ValidPassword {
+    String message() default "Repeat password don't match with password";
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/validator/ValidPassword.java]:
```suggestion
    String message() default "Repeat password doesn't match with password";
```

File: src/main/java/com/cinema/validator/PasswordValidator.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+import javax.validation.ConstraintValidatorContext;
+
+public class PasswordValidator implements ConstraintValidator<ValidPassword, UserRequestDto> {
+
+    @Override
+    public boolean isValid(UserRequestDto userRequestDto,
+                           ConstraintValidatorContext constraintValidatorContext) {
+        String password = userRequestDto.getPassword();
+        String repeatPassword = userRequestDto.getRepeatPassword();
+        return password != null && password.equals(repeatPassword);
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/validator/PasswordValidator.java]: This is optional: 
Objects.equals can be used as well in such cases

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**